packages/devtools_app/lib/src/network/network_controller.dart

```diff
@@ -146,15 +166,32 @@ class NetworkController {
   Future<void> startRecording({
     bool alreadyRecording = false,
   }) async {
-    final timestamp = await _networkService.updateLastRefreshTime(
-        alreadyRecording: alreadyRecording);
-
-    // Determine the offset that we'll use to calculate the approximate
-    // wall-time a request was made. This won't be 100% accurate, but it should
-    // easily be within a second.
-    _timelineMicrosOffset = DateTime.now().microsecondsSinceEpoch - timestamp;
-
-    await _toggleHttpTimelineRecording(true);
+    if (_httpRecordingNotifier.value || _startRecordingInProgress) {
+      // No need to start recording if we are already recording or we are in the
+      // process of starting recording.
+      return;
+    }
+    _httpRecordingNotifier.value = true;
+    _startRecordingInProgress = true;
+    try {
+      final timestamp = await _networkService.updateLastRefreshTime(
+          alreadyRecording: alreadyRecording);
+
+      // Determine the offset that we'll use to calculate the approximate
+      // wall-time a request was made. This won't be 100% accurate, but it should
+      // easily be within a second.
+      _timelineMicrosOffset = DateTime.now().microsecondsSinceEpoch - timestamp;
+      // TODO(jacobr): add an intermediate manager class to track which flags are
+      // set. We are setting more flags than we probably need to here but setting
+      // fewer flags risks breaking functionality on the timeline view that
+      // assumes that all flags are set.
+      await allowedError(serviceManager.service
```

Comment:
It might just be worth invoking `getVMTimelineFlags` first here to get the currently set flags. I doubt you'd see any performance hit from the extra RPC.

Reply:
yeah that would be fine from a performance perspective and is how I'd implement the client side flag manager in the TODO. We might want to add a bit of business logic to the VM in the future to simplify managing this further. For example, decouple it so setting timeline flags in one client doesn't impact other clients. Right now if you stopped recording the network events in one devtools instance you would break a second devtools instance that was also recording timeline events.